### PR TITLE
ci: fix test reports on forks (SDKCF-5616)

### PR DESCRIPTION
**Pull request by apparition47**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
       - 'APNS/**'
       - 'PusherMainView/**'
       - 'pusher/**'
-  pull_request:
+  pull_request_target:
```
Comment (imaginaris) [inline .github/workflows/test.yml:9]: @apparition47 Do you think it's worth to set `permissions:` just in case?

**Review by imaginaris — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (apparition47) [reply .github/workflows/test.yml:9]: Yes, but I can't figure out the combo of permissions this workflow needs. [It seems to use the GH upload artifact API](https://github.com/kishikawakatsumi/xcresulttool/blob/main/src/main.ts#L128) which I think requires the `actions:write` permission but I still can't get it to work.

Comment (imaginaris) [reply .github/workflows/test.yml:9]: Hmm. Did you try setting `permissions: write-all` temporarily to see if that helps?

Comment (apparition47) [reply .github/workflows/test.yml:9]: Doesn't `pull_request_target` already have `write-all` permissions?

Comment (imaginaris) [reply .github/workflows/test.yml:9]: Hmm, maybe. I'm not sure

Comment (apparition47) [reply .github/workflows/test.yml:4]: I think I got it working with `checks:write`

Comment (imaginaris) [reply .github/workflows/test.yml:4]: Great!

**Pull request merged**